packages/xsnap/src/xsnap.js

```diff
@@ -110,29 +116,52 @@ export async function xsnap(options) {
 
   assert(!/^-/.test(name), `name '${name}' cannot start with hyphen`);
 
-  /** @type {(() => Promise<void>) | undefined} */
-  let startCleanup;
-
   let args = [name];
-  await (snapshotStream &&
-    (async () => {
-      const tmpSnapPath = await ptmpName({
-        template: `load-snapshot-${safeHintFromDescription(
-          snapshotDescription,
-        )}-XXXXXX.xss`,
-      });
+  const postStart = await (snapshotStream &&
+    (snapshotUseFs
+      ? async () => {
```

Comment:
that's a pretty big trinary expression.. is there an easy way to define a pair of functions earlier and then make the `?:` fit on a single line?

Reply:
Did that and a little more